pickle5 wont build

I tried to install tripyview on levante into a clean environment on levante. I followed the steps in the Readme and got:


```
Building wheels for collected packages: pickle5, seawater
  Building wheel for pickle5 (setup.py) ... error
  error: subprocess-exited-with-error
  
  × python setup.py bdist_wheel did not run successfully.
  │ exit code: 1
  ╰─> [96 lines of output]
      running bdist_wheel
      running build
      running build_py
      creating build
      creating build/lib.linux-x86_64-cpython-311
      creating build/lib.linux-x86_64-cpython-311/pickle5
      copying pickle5/pickletools.py -> build/lib.linux-x86_64-cpython-311/pickle5
      copying pickle5/pickle.py -> build/lib.linux-x86_64-cpython-311/pickle5
      copying pickle5/__init__.py -> build/lib.linux-x86_64-cpython-311/pickle5
      creating build/lib.linux-x86_64-cpython-311/pickle5/test
      copying pickle5/test/test_picklebuffer.py -> build/lib.linux-x86_64-cpython-311/pickle5/test
      copying pickle5/test/test_pickle.py -> build/lib.linux-x86_64-cpython-311/pickle5/test
      copying pickle5/test/pickletester.py -> build/lib.linux-x86_64-cpython-311/pickle5/test
      copying pickle5/test/__init__.py -> build/lib.linux-x86_64-cpython-311/pickle5/test
      running build_ext
      building 'pickle5._pickle' extension
      creating build/temp.linux-x86_64-cpython-311
      creating build/temp.linux-x86_64-cpython-311/pickle5
      gcc -pthread -B /home/a/a270092/software/miniconda3/envs/tripyview/compiler_compat -DNDEBUG -fwrapv -O2 -Wall -fPIC -O2 -isystem /home/a/a270092/software/miniconda3/envs/tripyview/include -fPIC -O2 -isystem /home/a/a270092/software/miniconda3/envs/tripyview/include -fPIC -I/home/a/a270092/software/miniconda3/envs/tripyview/include/python3.11 -c pickle5/_pickle.c -o build/temp.linux-x86_64-cpython-311/pickle5/_pickle.o -std=c99
      pickle5/_pickle.c: In function 'Pdata_New':
      pickle5/_pickle.c:464:19: error: lvalue required as left operand of assignment
        464 |     Py_SIZE(self) = 0;
            |                   ^
      pickle5/_pickle.c: In function 'Pdata_clear':
      pickle5/_pickle.c:491:19: error: lvalue required as left operand of assignment
        491 |     Py_SIZE(self) = clearto;
            |                   ^
      pickle5/_pickle.c: In function 'Pdata_pop':
      pickle5/_pickle.c:542:23: error: lvalue required as decrement operand
        542 |     return self->data[--Py_SIZE(self)];
            |                       ^~
      pickle5/_pickle.c: In function 'Pdata_push':
      pickle5/_pickle.c:552:29: error: lvalue required as increment operand
        552 |     self->data[Py_SIZE(self)++] = obj;
            |                             ^~
      pickle5/_pickle.c: In function 'Pdata_poptuple':
      pickle5/_pickle.c:582:19: error: lvalue required as left operand of assignment
        582 |     Py_SIZE(self) = start;
            |                   ^
      pickle5/_pickle.c: In function 'Pdata_poplist':
      pickle5/_pickle.c:599:19: error: lvalue required as left operand of assignment
        599 |     Py_SIZE(self) = start;
            |                   ^
      pickle5/_pickle.c: In function '_Unpickler_ReadFromFile':
      pickle5/_pickle.c:1285:16: warning: implicit declaration of function '_PyObject_CallNoArg'; did you mean 'PyObject_CallNoArgs'? [-Wimplicit-function-declaration]
       1285 |         data = _PyObject_CallNoArg(self->readline);
            |                ^~~~~~~~~~~~~~~~~~~
            |                PyObject_CallNoArgs
      pickle5/_pickle.c:1285:14: warning: assignment to 'PyObject *' {aka 'struct _object *'} from 'int' makes pointer from integer without a cast [-Wint-conversion]
       1285 |         data = _PyObject_CallNoArg(self->readline);
            |              ^
      pickle5/_pickle.c: In function 'whichmodule':
      pickle5/_pickle.c:1937:15: warning: implicit declaration of function '_PySys_GetObjectId'; did you mean 'PySys_GetObject'? [-Wimplicit-function-declaration]
       1937 |     modules = _PySys_GetObjectId(&PyId_modules);
            |               ^~~~~~~~~~~~~~~~~~
            |               PySys_GetObject
      pickle5/_pickle.c:1937:13: warning: assignment to 'PyObject *' {aka 'struct _object *'} from 'int' makes pointer from integer without a cast [-Wint-conversion]
       1937 |     modules = _PySys_GetObjectId(&PyId_modules);
            |             ^
      pickle5/_pickle.c: In function 'save_float':
      pickle5/_pickle.c:2255:13: warning: implicit declaration of function '_PyFloat_Pack8'; did you mean 'PyFloat_Pack8'? [-Wimplicit-function-declaration]
       2255 |         if (_PyFloat_Pack8(x, (unsigned char *)&pdata[1], 0) < 0)
            |             ^~~~~~~~~~~~~~
            |             PyFloat_Pack8
      pickle5/_pickle.c: In function 'save':
      pickle5/_pickle.c:4452:30: warning: assignment to 'PyObject *' {aka 'struct _object *'} from 'int' makes pointer from integer without a cast [-Wint-conversion]
       4452 |                 reduce_value = _PyObject_CallNoArg(reduce_func);
            |                              ^
      pickle5/_pickle.c: In function 'load_binfloat':
      pickle5/_pickle.c:5428:9: warning: implicit declaration of function '_PyFloat_Unpack8'; did you mean 'PyFloat_Unpack8'? [-Wimplicit-function-declaration]
       5428 |     x = _PyFloat_Unpack8((unsigned char *)s, 0);
            |         ^~~~~~~~~~~~~~~~
            |         PyFloat_Unpack8
      pickle5/_pickle.c: In function 'load_pop':
      pickle5/_pickle.c:6178:30: error: lvalue required as left operand of assignment
       6178 |         Py_SIZE(self->stack) = len;
            |                              ^
      pickle5/_pickle.c: In function 'do_append':
      pickle5/_pickle.c:6535:42: error: lvalue required as left operand of assignment
       6535 |                     Py_SIZE(self->stack) = x;
            |                                          ^
      pickle5/_pickle.c:6541:34: error: lvalue required as left operand of assignment
       6541 |             Py_SIZE(self->stack) = x;
            |                                  ^
      pickle5/_pickle.c: In function 'load_additems':
      pickle5/_pickle.c:6663:38: error: lvalue required as left operand of assignment
       6663 |                 Py_SIZE(self->stack) = mark;
            |                                      ^
      pickle5/_pickle.c:6668:30: error: lvalue required as left operand of assignment
       6668 |         Py_SIZE(self->stack) = mark;
            |                              ^
      pickle5/_pickle.c: In function 'Pdata_pop':
      pickle5/_pickle.c:543:1: warning: control reaches end of non-void function [-Wreturn-type]
        543 | }
            | ^
      error: command '/sw/spack-levante/gcc-11.2.0-bcn7mb/bin/gcc' failed with exit code 1
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for pickle5
  Running setup.py clean for pickle5
  Building wheel for seawater (setup.py) ... done
  Created wheel for seawater: filename=seawater-3.3.4-py3-none-any.whl size=26454 sha256=6be3d2f6fa8ddb3a0929de94f850c074cd2771be7d7ed1737638e6992d854e5a
  Stored in directory: /home/a/a270092/.cache/pip/wheels/06/56/3a/dc3d7ed9c2cfed45162ece9ac117326ac711d5941c8cb42804
Successfully built seawater
Failed to build pickle5
ERROR: Could not build wheels for pickle5, which is required to install pyproject.toml-based projects
```
The error message suggests that there are issues with the compilation of the package's C extension modules. Why is not entirly clear to me. @patrickscholz Do you think this is a problem with pickle5 of with the specific environment that it's being installed in, or an incompatibility with the gcc compiler version used on levante?

Have you tried making a clean install recently?

--- a/README.md
+++ b/README.md
@@ -30,7 +30,7 @@ make sure your conda environment uses python=3.8
 # If you want to use the interactive features in tools/do_topo@elem_or_node.ipynb
 # due to python issues you will need jupyter_server=1.23.6, jupyter_client=7.3.2, 
 # tornad=6.1 --> its the only combination where this features works in moment!!!
-conda install -c conda-forge cartopy netcdf4=1.6.0
+conda install -c conda-forge cartopy netcdf4=1.6.0 python=3.8
 cd tripyview
 pip install -e .
 

